File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {

        Scanner sc = new Scanner(System.in);
        int M = sc.nextInt();
        int N = sc.nextInt();
        int[] arr = new int[N + 1];

        for (int i = 2; i <= N; i++) {
            arr[i] = i;
        }

        for (int i = 2; i <= Math.sqrt(N); i++) {
            if (arr[i] == 0) continue;

            for (int j = i + i; j <= N; j += i) {
                arr[j] = 0;
            }
        }

        for (int i = M; i <= N; i++) {
            if (arr[i] != 0) System.out.println(arr[i]);
        }
//        Arrays.stream(arr).filter(i -> i >= M).filter(i -> i != 0).forEach(System.out::println);

    }


}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
import sys
import math
input = sys.stdin.readline

M, N = map(int, input().split())

arr = [x for x in range(0, N + 1)]
arr[1] = 0

for i in range(2, int(math.sqrt(N) + 1)):
    if arr[i] == 0:
        continue
    for j in range(i + i, N + 1, i):
        arr[j] = 0

for i in arr:
    if i >= M and arr[i] != 0:
        print(arr[i])